activerecord/test/cases/finder_test.rb

```diff
@@ -1144,6 +1144,23 @@ def test_find_without_primary_key
       Matey.find(1)
     end
   end
+  
+  def test_first_and_last_with_limit_for_order_without_primary_key
+    # While Topic.first should impose an ordering by primary key,
+    # Topic.limit(n).first should not
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).first }
+    assert_sql(/^((?!ORDER).)*$/) { Post.limit(1).last }
```

Comment:
Let's remove the sql from these tests you've introduced, keep the other tests that assert SQL in place for this PR. If we want to change it, let's do it in a different PR to keep things focused.